src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs

```diff
@@ -85,6 +85,11 @@ public void Dispose()
         {
             DisposeRegistrations();
             _providerManager.Dispose();
+
+            foreach (IConfigurationSource source in _sources)
+            {
+                (source as IDisposable)?.Dispose();
```

Comment:
the manager has all the sources, so it can just iterate over them and dispose when possible